### ISSUE TITLE
[Request] Carryover under and over budget amounts.

### ISSUE THREAD

**Issue by Valiantiam** [inoda/ontrack#8]
The only large thing we are missing in this tool for it to be a hit in our household (forwarding this request from my GF) is if it could carryover over/under budgets. 

For example if we under-spent groceries, I want it to reflect that there is a bit more money available from the previous month, while still also showing what the actual budgeted monthly amount goal is. 

In the same vein, if we overspent on groceries category, it would show that although our goal is lets say 300, we overspent by 30 the previous month so its somehow reflected.

---

I struggle to decide how I'd prefer it to display, either right with each goal using some highlighting/extra lines, or maybe a separate tab that would give you insight to whether you are on track for 3m/6m/and 1y once averaged out depending on the category. 

Im not good at mock design or I'd show, but I'd be happy to provide any more details that might be helpfu,.

**Comment by Valiantiam** [comment]
@inoda

**Comment by inoda** [comment]
@Valiantiam So for full transparency - my wife designed this app :D Her thinking was that it'd be nice to carryover over-spend, but for under-spend the money should just be put away and forgotten about. What do you think about this? I agree that seeing some sort of aggregate of over-spending would be very useful. **Maybe a sum of over-spending minus sum of under-spending as a status somewhere?**

Also, she is currently job hunting/interviewing and is quite busy so I can't promise a speedy implementation of this.

**Comment by Valiantiam** [comment]
@inoda yeah an overall sum but organized by category would be fine too. 

I agree with the notion of that money which is underspent just being put away, and I actually do that. 

I'd like to track however for example; I do a lot of trips during the summer where I purchase extra groceries and alcohol [more than usual ;) ]  and it'd be nice to be able to see if overall I'm budgeting appropriately throughout the year by under spending on some months to make up for overspending on others.

Also totally understand the job hunt/life priorities. Just figured Id give it a shot of making the suggestion. I can look more at the code directly and perhaps offer some small pull requests though I'm not a programmer at heart. 

I appreciate you looking at my request and responding nonetheless, and let your wife know good luck on the job hunt.

**Comment by inoda** [comment]
@Valiantiam Yeah that totally makes sense! I've been wanting something similar I think, to understand how we're doing overall rather than just month to month. The annual bar chart is helpful but doesn't contain that over/under spend.

I will try adding an over/under spend by category section on the insights page some time in the next couple of weeks!
